TestCases/compliance-level-3/0067-feel-split-function/0067-feel-split-function.dmn

```diff
@@ -0,0 +1,110 @@
+    <description>FEEL built-in function 'split(string,delimiter)' in category string functions</description>
+
+    <itemDefinition name="tStringList" isCollection="true">
+        <typeRef>string</typeRef>
+    </itemDefinition>
+
+    <decision name="decision001" id="_decision001">
+        <description>Tests FEEL expression: 'split("John Doe", "\s")' and expects result: '["John", "Doe"] (list)'</description>
+        <question>Result of FEEL expression 'split("John Doe", "\s")'?</question>
+        <allowedAnswers>["John", "Doe"] (list)</allowedAnswers>
+        <variable typeRef="tStringList" name="decision001"/>
+        <literalExpression>
+            <text>split("John Doe", "\s")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision001_a" id="_decision001_a">
+        <description>Tests FEEL expression: 'split("a;b;c;;", ";")' and expects result: '["a", "b", "c", "", ""] (list)'</description>
+        <question>Result of FEEL expression 'split(10, -4)'?</question>
+        <allowedAnswers>["a", "b", "c", "", ""]</allowedAnswers>
+        <variable typeRef="tStringList" name="decision001_a"/>
+        <literalExpression>
+            <text>split("a;b;c;;", ";")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision004" id="_decision004">
+        <description>Tests FEEL expression: 'split()' and expects result: 'null (list)'</description>
+        <question>Result of FEEL expression 'split()'?</question>
+        <allowedAnswers>null (list)</allowedAnswers>
+        <variable typeRef="tStringList" name="decision004"/>
+        <literalExpression>
+            <text>split()</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision005" id="_decision005">
+        <description>Tests FEEL expression: 'split("foo")' and expects result: 'null (list)'</description>
+        <question>Result of FEEL expression 'split("foo")'?</question>
+        <allowedAnswers>null (list)</allowedAnswers>
+        <variable typeRef="tStringList" name="decision005"/>
+        <literalExpression>
+            <text>split("foo")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision005_a" id="_decision005_a">
+        <description>Tests FEEL expression: 'split("foo","f","f")' and expects result: 'null (list)'</description>
+        <question>Result of FEEL expression 'split("foo","f","f")'?</question>
+        <allowedAnswers>null (list)</allowedAnswers>
+        <variable typeRef="tStringList" name="decision005_a"/>
+        <literalExpression>
+            <text>split("foo","f","f")</text>
```

Comment:
I think `split()` built-in function was discussed at some point, and at least in my perspective the DMNv1.2 Beta1 spec is missing to specify a flag parameter in order to make it actually usable. I need to check what is the issue# or eventually raise one. I would propose to drop this test case based on the fact it would prevent any vendor for implementing in such a strategy, at least until clarified.

Reply:
Thanks @tarilabs.  The spec has no flags on it but that doesn't mean it is unusable or unimplementable. It works without flags for many cases as the test show.  

Might I suggest we keep the tests - they verify what the 1.2 spec says.  If flags are added in a later spec the tests can be updated, but the test verify what can be used.

Reply:
@StrayAlien in this case, the missing flag on split() was actually my fault (yeah, this is one of the mistakes I mentioned). Without the support for flags it is not possible to split multi-line strings for instance, or ignore case, etc. It was an oversight that needs to be fixed on DMN 1.3. 
See functions matches() and replace() that have the optional "flags" parameter in place. 

In regards to the TCK, we can add a test that effectively prevents vendors from implementing the optional "flags" parameter (what you are proposing here) or we can omit such test and let vendors that want to offer such feature in advance of DMN 1.3 to do so. 

Our approach in the past in cases like this was to add positive tests for such "upcoming" features in the non-compliant folder and wait for the next DMN version to move them to the compliant one. For example, we did the same for functions any() and all() that were known to come in DMN 1.2 to replace the previously named functions or() and and(). Vendors that wished to do so, could implement the feature in advance, without penalty or prejudice.

The TCK is community driven, so we can always decide to change our minds and do different from now on. Being obviously biased (I want to enable users to do what they need), I usually defer the decision to the group (/cc @agilepro) and will abide by such.

Reply:
Thanks @etirelli, none of us are perfect - just look at some of my spec interpretations! :-)

In fact, to further prove it - I have only really just understood what @tarilabs  really meant.  Doh!  The test is _preventing_ 3 params.  Sorry - I just totally missed that point.  I was, after all, doing this during my lunch while stuffing my face.  :-)

The 'too many params' assertion is just a standard thing I did for all funcs I tested.  I think you are right, if it is coming in a future spec then I think it easiest just to remove that assertion for this function.

No problems.  Apologies to you and @tarilabs for not quite getting it sooner!